py/desispec/zfind/redmonster.py

```diff
@@ -58,14 +59,35 @@ def __init__(self, wave, flux, ivar, R=None, dloglam=1e-4):
         self.nwave = nwave
         self.nspec = nspec
 
+        #- Standardize objtype, converting ELG,LRG -> GALAXY, make upper case
+        templatetypes = set()
+        if objtype is None:
+            templatetypes = set(['GALAXY', 'STAR', 'QSO'])
+        else:
+            if isinstance(objtype, str):
+                objtype = [objtype,]
+                
+            objtype = [x.upper() for x in objtype]
+            for x in objtype:
+                if x in ['ELG', 'LRG']:
+                    templatetypes.add('GALAXY')
+                elif x in ['QSO', 'GALAXY', 'STAR']:
+                    templatetypes.add(x)
+                else:
+                    raise ValueError('Unknown objtype '+x)
+            
         #- list of (templatename, zmin, zmax) to fix
         self.template_dir = os.getenv('REDMONSTER_TEMPLATES_DIR')
-        self.templates = [
-            ('ndArch-spEigenStar-55734.fits', -0.005, 0.005),
-            ('ndArch-ssp_em_galaxy-v000.fits', 0.6, 1.6),
-            # ('ndArch-ssp_em_galaxy_quickdesi-v000.fits', 0.6, 1.6),
-            ('ndArch-QSO-V003.fits', 0.0, 3.5),
-        ]
+        self.templates = list()
+        for x in templatetypes:
+            if x == 'GALAXY':
+                self.templates.append(('ndArch-ssp_em_galaxy-v000.fits', 0.6, 1.6))
+            elif x == 'STAR':
+                self.templates.append(('ndArch-spEigenStar-55734.fits', -0.005, 0.005))
+            elif x == 'QSO':
+                self.templates.append(('ndArch-QSO-V003.fits', 0.0, 3.5))
+            else:
```

Comment:
Whoop!  Good catch.  Redmonster doesn't distinguish between ELGs and LRGs for its templates, but this code is restricting the GALAXY=(ELG or LRG) search to a ELG-ish redshift range of 0.6 to 1.6.  At minimum we should include a range appropriate to both and a superset of what might be in the targets rather than what we want to be in the targets.  However, if a future redmonster version (or alternate redshift fitter) had ELG-optimized templates, I'd be suspicious of answers with z>1.67 since [OII] has walked off the edge of the spectrograph and there wouldn't be much real signal to latch onto.  How about [0,1.67] for this range?  Or should it really be out to z=2?